src/Tizen.NUI.Components/Controls/ViewList.cs

```diff
@@ -0,0 +1,302 @@
+            /// </summary>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public void Clear()
+            {
+                viewList.Select(-1);
+                for (int i = 0; i < internalViewList.Count; i++)
+                {
+                    viewList.RemoveAt(0);
+                }
+                internalViewList.Clear();
+            }
+            /// <summary>
+            /// Checks is a View object is in the list.
+            /// </summary>
+            /// <returns>Whether the list contains the View object</returns>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public bool Contains(View view)
+            {
+                return internalViewList.Contains(view);
+            }
+
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public void CopyTo(View[] array, int arrayIndex)
+            {
+            }
+            /// <summary>
+            /// Remove a View from the list.
+            /// </summary>
+            /// <param name="view">The to-be-removed View object</param>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public bool Remove(View view)
+            {
+                if (!internalViewList.Contains(view)) return false;
+                RemoveAt(internalViewList.IndexOf(view));
+                return true;
+            }
+            /// <summary>
+            /// Get the enumerator of the list.
+            /// </summary>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            public IEnumerator<View> GetEnumerator()
+            {
+                return internalViewList.GetEnumerator();
+            }
+            /// <summary>
+            /// Get the enumerator of the list.
+            /// </summary>
+            [EditorBrowsable(EditorBrowsableState.Never)]
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return internalViewList.GetEnumerator();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Frame is a class which keeps Views in a frame and
+    /// shows one selected View.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Frame : ViewList
+    {
+        private View frameView;
+
+        /// <summary>
+        /// Creates a new instance of Frame.
+        /// </summary>
+        public Frame()
+        {
+            frameView = new View();
+            frameView.WidthSpecification = LayoutParamPolicies.MatchParent;
+            frameView.HeightSpecification = LayoutParamPolicies.MatchParent;
+            this.Add(frameView);
+        }
+
+        /// <summary>
+        /// Insert a View to the Frame.
+        /// </summary>
+        /// <param name="order"></param>
+        /// <param name="tab"></param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void Insert(int order, View view)
+        {
+            view.WidthSpecification = LayoutParamPolicies.MatchParent;
```

Comment:
Please validate argument. if `view` is null, it will throw `NullReferenceException`.